pkg/core/native/policy.go

```diff
@@ -0,0 +1,353 @@
+func (p *Policy) GetFeePerByteInternal(ic *interop.Context) *big.Int {
+	return p.getBigIntWithKey(ic, feePerByteKey)
+}
+
+// getBlockedAccounts is Policy contract method and returns list of blocked
+// accounts hashes.
+func (p *Policy) getBlockedAccounts(ic *interop.Context, _ []stackitem.Item) stackitem.Item {
+	ba, err := p.GetBlockedAccountsInternal(ic)
+	if err != nil {
+		panic(err)
+	}
+	return ba.ToStackItem()
+}
+
+// GetBlockedAccountsInternal returns list of blocked accounts hashes.
+func (p *Policy) GetBlockedAccountsInternal(ic *interop.Context) (BlockedAccounts, error) {
+	si := ic.DAO.GetStorageItem(p.Hash, blockedAccountsKey)
+	if si == nil {
+		return nil, errors.New("BlockedAccounts uninitialized")
+	}
+	ba, err := BlockedAccountsFromBytes(si.Value)
+	if err != nil {
+		return nil, err
+	}
+	return ba, nil
+}
+
+// setMaxTransactionsPerBlock is Policy contract method and  sets the upper limit
+// of transactions per block.
+func (p *Policy) setMaxTransactionsPerBlock(ic *interop.Context, args []stackitem.Item) stackitem.Item {
+	if ok, err := p.checkValidators(ic); err != nil {
+		panic(err)
+	} else if !ok {
+		return stackitem.NewBool(false)
+	}
+	value := toBigInt(args[0])
+	if err := p.setBigIntWithKey(ic, maxTransactionsPerBlockKey, value); err != nil {
+		panic(err)
+	}
+	return stackitem.NewBool(true)
+}
+
+// setMaxBlockSize is Policy contract method and sets maximum block size.
+func (p *Policy) setMaxBlockSize(ic *interop.Context, args []stackitem.Item) stackitem.Item {
+	if ok, err := p.checkValidators(ic); err != nil {
+		panic(err)
+	} else if !ok {
+		return stackitem.NewBool(false)
+	}
+	value := toBigInt(args[0])
+	if payload.MaxSize <= value.Int64() {
+		return stackitem.NewBool(false)
+	}
+	if err := p.setBigIntWithKey(ic, maxBlockSizeKey, value); err != nil {
+		panic(err)
+	}
+	return stackitem.NewBool(true)
+}
+
+// setFeePerByte is Policy contract method and sets transaction's fee per byte.
+func (p *Policy) setFeePerByte(ic *interop.Context, args []stackitem.Item) stackitem.Item {
+	if ok, err := p.checkValidators(ic); err != nil {
+		panic(err)
+	} else if !ok {
+		return stackitem.NewBool(false)
+	}
+	value := toBigInt(args[0])
+	if err := p.setBigIntWithKey(ic, feePerByteKey, value); err != nil {
+		panic(err)
+	}
+	return stackitem.NewBool(true)
+}
+
+// blockAccount is Policy contract method and adds given account hash to the list
+// of blocked accounts.
+func (p *Policy) blockAccount(ic *interop.Context, args []stackitem.Item) stackitem.Item {
+	if ok, err := p.checkValidators(ic); err != nil {
+		panic(err)
+	} else if !ok {
+		return stackitem.NewBool(false)
+	}
+	value := toUint160(args[0])
+	si := ic.DAO.GetStorageItem(p.Hash, blockedAccountsKey)
+	if si == nil {
+		panic("BlockedAccounts uninitialized")
+	}
+	ba, err := BlockedAccountsFromBytes(si.Value)
+	if err != nil {
+		panic(err)
+	}
+	if len(ba) != 0 {
+		i := sort.Search(len(ba), func(i int) bool {
+			return ba[i].Equals(value)
```

Comment:
I don't think this is correct way to use `sort.Search` as it assumes second argument is monotonic `f(i) == true -> f(i+1) == true`.
https://golang.org/pkg/sort/#Search
This said we can use `!ba[i].Less(value)` and compare found item (if any) on equality.

Reply:
That's true, got rid of this. `!ba[i].Less(value)` also helps to remove the second `Search`.